src/Sentry/SentryClient.cs

```diff
@@ -121,22 +121,68 @@ public void CaptureTransaction(Transaction transaction)
                                 "to properly finalize the transaction and send it to Sentry.");
         }
 
-
         // Sampling decision MUST have been made at this point
-        Debug.Assert(transaction.IsSampled != null,
+        Debug.Assert(transaction.IsSampled is not null,
             "Attempt to capture transaction without sampling decision.");
 
-
-        if (transaction.IsSampled != true)
+        if (transaction.IsSampled is false)
         {
             _options.ClientReportRecorder.RecordDiscardedEvent(DiscardReason.SampleRate, DataCategory.Transaction);
             _options.LogDebug("Transaction dropped by sampling.");
             return;
         }
 
+        var processedTransaction = BeforeSendTransaction(transaction);
+        if (processedTransaction is null) // Rejected transaction
+        {
+            _options.ClientReportRecorder.RecordDiscardedEvent(DiscardReason.BeforeSend, DataCategory.Transaction);
+            _options.LogInfo("Transaction dropped by BeforeSendTransaction callback.");
+            return;
+        }
+
         CaptureEnvelope(Envelope.FromTransaction(transaction));
```

Comment:
I've just noticed this bit when working on Profiling - this should be sending `processedTransaction` instead

Reply:
Yep!  @SeanFeldman - sorry I missed that in my review.   I think you can fix in a new PR.  Thanks.

Reply:
Thank you for catching this, @vaind.
@mattjohnsonpint, I'll send a follow up PR.

Reply:
Follow-up PR: #2204